File: Docs/Reference/Comparisons/rubymotion_demo.m
@interface HelloView : UIView {
    BOOL _hasMoved;
    NSUInteger _touchCount;
}
@end

@implementation HelloView
- (void)drawRect:(NSRect)dirtyRect
{
    UIColor *bgColor;
    NSString *text;
    if(_hasMoved) {
        bgColor = [UIColor colorWithRed: [@0.0 randUpTo: @1.0]
                                  green: [@0.0 randUpTo: @1.0]
                                   blue: [@0.0 randUpTo: @1.0]];
        text = @"ZOMG!";
    } else {
        bgColor = [UIColor blackColor];
        if(_touchCount > 0)
            text = [NSString stringWithFormat:@"Touched %ld times!", _touchCount];
        else
            text = @"Hello world!";
    }
    [bgColor set];
    [[UIBezierPath bezierPathWithRect: dirtyRect] fill];

    [[UIColor whiteColor] set];
    [text drawAtPoint:(CGPoint) {10, 20} withFont:[UIFont systemFontOfSize: 24]];
}
@end

File: Docs/Reference/Comparisons/rubymotion_demo.rb
class HelloView < UIView
    def drawRect(rect)
        if @moved
            bgColor = begin
                red, green, blue = rand(100), rand(100), rand(100)
                UIColor colorWithRed(red/100.0, green:green/100.0, blue:blue/100.0, alpha: 1.0)
            end
            text = "ZOMG!"
        else
            bgColor = UIColor.blackColor
            text = @touches ? "Touched #{@touches} times!" : "Hello RubyMotion!"
        end
        bgColor.set
        UIBezierPath.bezierPathWithRect(rect).fill
        UIColor.whiteColor.set
        text.drawAtPoint(CGPointMake(10, 20), withFont:UIFont.systemFontOfSize(24))
    end
end
